src/Transformer/AbstractEnumTransformer.php

```diff
@@ -0,0 +1,48 @@
+<?php
+declare(strict_types=1);
+
+namespace Diaclone\Transformer;
+
+use Diaclone\Resource\ResourceInterface;
+
+abstract class AbstractEnumTransformer extends AbstractTransformer
+{
+    public function getPropertyValue($data, $property)
+    {
+        if (empty($property)) {
+            return $data;
+        }
+
+        if (is_array($data)) {
+            return $data[$property] ?? null;
+        }
+
+        // convert property to camelCase
+        $getter = 'get' . str_replace(' ', '', ucwords(str_replace(['-', '_'], ' ', $property)));
+        if (method_exists($data, $getter)) {
+            $enum = $data->$getter();
+            return $enum->getValue();
+        }
+
+        return $data->$property ?? null;
+    }
+    
+    public function transform(ResourceInterface $resource)
+    {
+        return $this->getPropertyValueFromResource($resource);
+    }
+
+    public function untransform(ResourceInterface $resource)
+    {
+        if ($resource->getData() === null) {
+            return null;
+        }
+
+        $data = $resource->getData();
+        $propertyTransform = $this->getObjectClass();
+
+        return $propertyTransform::byValue($data);
```

Comment:
You are right, it is an unsafe method, what do you think if we implement that approach:

```
$enumObject = null;
try {
   $enumObject = $propertyTransform::byValue($data);
} catch ( InvalidArgumentException $e) {
   $enumObject = '';
}
return $enumObject;
```
Or by using `Enum::getConstants()` you are doing extra stuff like send a custom message when the value is not defined?